src/plugins/intel_npu/src/plugin/npuw/just_sync_infer_request.cpp

```diff
@@ -919,6 +920,29 @@ void ov::npuw::JustInferRequest::unsafe_run_this_prep_next(std::size_t idx, bool
     }  // if (replaced_by)
 }
 
+ov::Tensor ov::npuw::JustInferRequest::allocTensor(const ov::element::Type type,
+                                                   const ov::Shape& shape,
+                                                   const std::string& device) {
+    if (!m_alloc_required || device == "CPU") {
+        return ov::Tensor(type, shape);
+    }
+    if (std::any_of(shape.begin(), shape.end(), [&](std::size_t dim) {
+            return dim == 0;
+        })) {
+        return ov::Tensor(type, shape);
+    }
+
+    ov::SoPtr<ov::ITensor> remote_tensor;
+    ov::Tensor allocated_tensor;
+    {
+        std::lock_guard<std::mutex> guard(m_alloc_mutex);
+        m_remote_ctx = m_npuw_model->get_plugin()->get_core()->get_default_context(device)._ptr;
+        remote_tensor = m_remote_ctx->create_host_tensor(type, shape);
+        allocated_tensor = ov::make_tensor(remote_tensor);
```

Comment:
why do you need a mutex here? you call `allocTensor` from multiple threads?

Reply:
Just in case since we do guard allocation in banks